docs/design/aws_under_the_hood.md

```diff
@@ -0,0 +1,271 @@
+[here](http://releases.k8s.io/release-1.0/docs/design/aws_under_the_hood.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Peeking under the hood of kubernetes on AWS
+
+We encourage you to use kube-up (or CloudFormation) to create a cluster.  But
+it is useful to know what is being created: for curiosity, to understand any
+problems that may arise, or if you have to create things manually because the
+scripts are unsuitable for any reason.  We don't recommend manual configuration
+(please file an issue and let us know what's missing if there's something you
+need) but sometimes it is the only option.
+
+This document sets out to document how kubernetes on AWS maps to AWS objects.
+Familiarity with AWS is assumed.
+
+### Top-level
+
+Kubernetes consists of a single master node, and a collection of minion nodes.
+Other documents describe the general architecture of Kubernetes (all nodes run
+Docker; the kubelet agent runs on each node and launches containers; the
+kube-proxy relays traffic between the nodes etc).
+
+By default on AWS:
+
+* Instances run Ubuntu 15.04 (the official AMI).  It includes a sufficiently
+  modern kernel to give a good experience with Docker, it doesn't require a
+  reboot.  (The default SSH user is `ubuntu` for this and other ubuntu images)
+* By default we run aufs over ext4 as the filesystem / container storage on the
+  nodes (mostly because this is what GCE uses).
+
+These defaults can be changed by passing different environment variables to
+kube-up.
+
+### Storage
+
+AWS does support persistent volumes via EBS.  These can then be attached to
+pods that should store persistent data (e.g. if you're running a database).
+
+Minions do not have persistent volumes otherwise.  In general, kubernetes
+containers do not have persistent storage unless you attach a persistent
+volume, and so minions on AWS use instance storage.  Instance storage is
+cheaper, often faster, and historically more reliable.  This does mean that you
+should pick an instance type that has sufficient instance storage, unless you
+can make do with whatever space is left on your root partition.
+
+The master _does_ have a persistent volume attached to it.  Containers are
+mostly run against instance storage, just like the minions, except that we
+repoint some important data onto the peristent volume.
+
+By default we use aufs over ext4.  `DOCKER_STORAGE=btrfs` is also a good choice
+for a filesystem: it is relatively reliable with Docker; btrfs itself is much
+more reliable than it used to be with modern kernels.  It can easily span
+multiple volumes, which is particularly useful when we are using an instance
+type with multiple ephemeral instance disks.
+
+### AutoScaling
+
+We run the minions in an AutoScalingGroup.  Currently auto-scaling (e.g. based
```

Comment:
Is there a way to make the master/minion distinction here?  Minions are run in an ASG, but the master currently is not.  I want to move the master to an ASG of size 1 so that it auto-recovers, but the fact that the master will not automatically recover is currently an important gotcha (that I need to fix)